src/main/java/org/candlepin/pki/impl/JSSProviderLoader.java

```diff
@@ -122,43 +74,20 @@ public static synchronized void initialize() {
         log.info("Initializing JSS CryptoManager...");
         log.info("Using JSS v{}", getJSSVersion());
 
-        ClassLoader loader = JSSProviderLoader.class.getClassLoader();
+        InitializationValues initValues = new InitializationValues(NSS_DB_LOCATION);
 
-        Object initializationValuesObject = getInitializationValuesObject(loader);
-        Class<?> ivsClass = initializationValuesObject.getClass();
+        initValues.noCertDB = true;
+        initValues.readOnly = false;
+        initValues.noModDB = false;
+        initValues.installJSSProvider = false;
+        initValues.initializeJavaOnly = false;
 
         try {
-            // Set values on fields of the InitializationValues object.
-            Field noCertDB = ivsClass.getField("noCertDB");
-            Field readOnly = ivsClass.getField("readOnly");
-            Field noModDB = ivsClass.getField("noModDB");
-            Field installJSSProvider = ivsClass.getField("installJSSProvider");
-            Field initializeJavaOnly = ivsClass.getField("initializeJavaOnly");
-
-            noCertDB.set(initializationValuesObject, true);
-            readOnly.set(initializationValuesObject, false);
-            noModDB.set(initializationValuesObject, false);
-            installJSSProvider.set(initializationValuesObject, false);
-            initializeJavaOnly.set(initializationValuesObject, false);
-
-            // Initialize the CryptoManager, which will initialize the nss DB.
-            CryptoManager.class.getMethod("initialize", ivsClass)
-                .invoke(null, initializationValuesObject);
+            CryptoManager.initialize(initValues);
         }
-        catch (NoSuchMethodException | NoSuchFieldException | InvocationTargetException |
-            IllegalAccessException e) {
-
-            if (e.getCause() instanceof AlreadyInitializedException) {
-                log.warn("CryptoManager was already initialized.");
-            }
-            else if (e.getCause() instanceof KeyDatabaseException ||
-                e.getCause() instanceof CertDatabaseException ||
-                e.getCause() instanceof GeneralSecurityException) {
-                throw new JSSLoaderException("Could not initialize CryptoManager!", e.getCause());
-            }
-            else {
-                throw new JSSLoaderException("Could not initialize CryptoManager!", e);
-            }
+        catch (AlreadyInitializedException | CertDatabaseException | GeneralSecurityException |
+            KeyDatabaseException e) {
+            throw new JSSLoaderException("Could not initialize CryptoManager!", e);
```

Comment:
`AlreadyInitializedException` shouldn't result in an exception, only a warning

Reply:
Should also use this opportunity to move the InitializationValues bits into the try block for better organization.

Reply:
Yes I tested in both 8 and 9 (both downgrade and Java17).

I think the exception here is correct. `AlreadyInitializedException` should never happen because we handle that case at the beginning of the method. Warning would leave us in invalid state.

@Ceiu Why? Putting simple assignmnets into try block will only obscure the single line that actually throws something.

Reply:
Technically it should never happen at the moment, but code changes, so if for whatever reason that check is removed in the future, and something tries to re-initialize the cryptomanager, `AlreadyInitializedException` will be thrown. And if that happens there is no invalid state afaik. Something tried to re-initialize, it failed, but everything still works, so a warning is better instead of throwing a catastrophic error that will stop candlepin from deploying

Reply:
Even in that case an error would make sense. Candlepin would would fail according to fail-fast phylosophy as broken code was introduced.

Reply:
Fair enough, failing fast sounds good

Reply:
@Januson 
Stylistic choice, I guess. In my view, you can't do the targeted op without those initialization values -- they're basically arguments for it -- so they logically live/belong with that statement like the arguments of any typical function call. Additionally, they're not used outside of it, so my preference is to block them up accordingly.